nifi-nar-bundles/nifi-graph-bundle/nifi-other-graph-services/src/main/java/org/apache/nifi/graph/TinkerpopClientService.java

```diff
@@ -0,0 +1,545 @@
+            traversalSourceName = context.getProperty(TRAVERSAL_SOURCE_NAME).evaluateAttributeExpressions()
+                    .getValue();
+        }
+
+        scriptSubmission = context.getProperty(SUBMISSION_TYPE).getValue().equals(SCRIPT_SUBMISSION.getValue());
+
+        cluster = buildCluster(context);
+    }
+
+    @OnDisabled
+    public void shutdown() {
+        try {
+            compiledCode = null;
+            engine = null;
+            if (traversalSource != null) {
+                traversalSource.close();
+            }
+        } catch (Exception e) {
+            throw new ProcessException(e);
+        } finally {
+            if (cluster != null) {
+                cluster.close();
+            }
+            cluster = null;
+            traversalSource = null;
+        }
+    }
+
+    @Override
+    public Map<String, String> executeQuery(String s, Map<String, Object> map, GraphQueryResultCallback graphQueryResultCallback) {
+        try {
+            if (scriptSubmission) {
+                return scriptSubmission(s, map, graphQueryResultCallback);
+            } else {
+                return bytecodeSubmission(s, map, graphQueryResultCallback);
+            }
+        } catch (Exception ex) {
+            throw new ProcessException(ex);
+        }
+    }
+
+    @Override
+    public String getTransitUrl() {
+        return this.transitUrl;
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return DESCRIPTORS;
+    }
+
+    @Override
+    public Collection<ValidationResult> customValidate(ValidationContext context) {
+        Collection<ValidationResult> results = new ArrayList<>();
+        boolean jarsIsSet = !StringUtils.isEmpty(context.getProperty(EXTRA_RESOURCE).getValue());
+        boolean clzIsSet = !StringUtils.isEmpty(context.getProperty(EXTENSION_CLASSES).getValue());
+
+        if (jarsIsSet && clzIsSet) {
+            try {
+                final ClassLoader loader = ClassLoaderUtils
+                        .getCustomClassLoader(context.getProperty(EXTRA_RESOURCE).getValue(), this.getClass().getClassLoader(), null);
+                String[] classes = context.getProperty(EXTENSION_CLASSES).evaluateAttributeExpressions().getValue().split(",[\\s]*");
+                for (String clz : classes) {
+                    Class.forName(clz, true, loader);
+                }
+            } catch (Exception ex) {
+                results.add(new ValidationResult.Builder().subject(EXTENSION_CLASSES.getDisplayName()).valid(false).explanation(ex.toString()).build());
+            }
+        }
+
+        boolean standardConfigIsSet = context.getProperty(CONTACT_POINTS).isSet()
+                && context.getProperty(PATH).isSet()
+                && context.getProperty(PORT).isSet();
+        boolean fileIsSet = context.getProperty(REMOTE_OBJECTS_FILE).isSet();
+
+        if (standardConfigIsSet && fileIsSet) {
```

Comment:
Do we need all of these if the properties are dependent?

Reply:
This was erroneous. Removed several of the validation checks.